diff --git a/external/njs_xml_module.c b/external/njs_xml_module.c
index 0baaea54..a2be4d7f 100644
--- a/external/njs_xml_module.c
+++ b/external/njs_xml_module.c
@@ -1986,10 +1986,10 @@ njs_xml_nset_cleanup(void *data)
 static void
 njs_xml_error(njs_vm_t *vm, njs_xml_doc_t *current, const char *fmt, ...)
 {
-    u_char         *p, *last;
-    va_list        args;
-    xmlError       *err;
-    u_char         errstr[NJS_MAX_ERROR_STR];
+    u_char          *p, *last;
+    va_list         args;
+    const xmlError  *err;
+    u_char          errstr[NJS_MAX_ERROR_STR];
 
     last = &errstr[NJS_MAX_ERROR_STR];
 
